shared/dateSync.js

```diff
@@ -0,0 +1,36 @@
+/*
+- When storing dates, do it based on UTC time.
+- When displaying a date to the user, use convertDateToLocale().
+- If you need to convert a date back to UTC, use convertDateToUTC
+
+// NOTE: Half done to close #531
+*/
+
+// Create date always based on UTC time
+export function createDateUTC (date) {
+  const now = date ? new Date(date) : new Date()
+  const offset = now.getTimezoneOffset()
+  const minutes = now.getMinutes()
+  now.setMinutes(minutes + offset)
+
+  return now
+}
+
+// Convert UTC date to date based on Locale
+// ex: 1570599537334 (06:38:57) (UTC)
+// ->  1570603137334 (07:38:57) (locale, +01 WEST)
+export function convertDateToLocale (date) {
+  const now = new Date(date)
+  const offset = now.getTimezoneOffset()
+  const minutes = now.getMinutes()
+  now.setMinutes(minutes - offset)
+
+  return now
+}
+
+// Convert Locale date to date based on UTC
+// ex: 1570603137334 (07:38:57) (locale, +01 WEST)
+// ->  1570599537334 (06:38:57) (UTC time)
+export function convertDateToUTC (date) {
+  return createDateUTC(date)
+}
```

Comment:
It's unclear to me why this file was created, and why ISO strings were deviated from. This does not seem like a simplification to me.

I chose to use ISO strings after looking at this post, which makes a strong case for them.

- https://zachholman.com/talk/utc-is-enough-for-everyone-right

Closing #531 doesn't require deviation from the use of ISO strings, rather, it just requires a backend API to return an ISO string (represent the server's current time).

Also, besides being an unnecessary complication of the existing implementation, this code contains a serious bug because it conducts math on `getTimezoneOffset()` with `getMinutes()`, and uses that to `setMinutes()`, [which only takes a value from `0-59`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setMinutes):

> An integer between 0 and 59, representing the minutes.

Whereas the resulting math will be out of range. So this file should be deleted, and the time stuff should be reverted.

Reply:
reverted to ISO